### DISCUSSION TITLE
Insert_text and and Adobe Acrobat DC has encountered an unknown error.

### DISCUSSION THREAD

**Discussion by Avtonkadov** [pymupdf/PyMuPDF#2121] [Looking for help] [answered]
With PyMuPDF I add text to pdf, but then if open output file with Adobe Acrobat and try to edit this file, it massage  error.

![Error](https://user-images.githubusercontent.com/120422847/207160908-7308b955-0c21-4374-8910-7a418833d0e7.PNG)

```
import io
import os
import fitz
doc = fitz.open("in.pdf")
ffile = "C:/windows/fonts/times.ttf"  
text = "Раздел 22. Book 1. Chapter 2.pdf"
for page in doc:
    w=page.rect.width
    h=page.rect.height
    page.clean_contents()
    page.insert_font (fontname="f0", fontfile=ffile, encoding=fitz.TEXT_ENCODING_CYRILLIC)
    if page.rotation==270:
        ww = int(w-190/0.352)
        #excess = page.insert_textbox(fitz.Rect(0, ww, 14,ww+300), text+"iijij",fontname="f0", fontsize = 12, rotate=-90)
        excess = page.insert_text(fitz.Point(5, ww), text,fontname="f0", fontsize = 12, rotate=-90)
    else:
        ww = int(w-190/0.352)
        #excess = page.insert_textbox(fitz.Rect(ww, h-14, ww+300,h+14), text,fontname="f0", fontsize = 12, rotate=0)
        excess = page.insert_text(fitz.Point(ww, h-5), text, fontname="f0", fontsize = 12, rotate=0)
doc.ez_save( "out.pdf")
print (doc.page_count)
doc.close()
```

[This is video, how it works](https://www.youtube.com/watch?v=eYx5hUtGdT8)

This is input file. Input file from Autodesk Autocad (dwg to pdf.pc3) 
[in.pdf](https://github.com/pymupdf/PyMuPDF/files/10212196/in.pdf)
This is output file
[out.pdf](https://github.com/pymupdf/PyMuPDF/files/10212197/out.pdf)

**Comment by JorjMcKie** [comment, marked as answer]
You did not mention your PyMuPDF version.
Presumably your problem has to do with an issue fixed in the most current release 1.21.1rc1.
I checked your script and reproducer PDF and no problem occurred.

**Reply by Avtonkadov** [reply]
Thank You very much!

**Comment by yang-521** [comment]
I also encountered this problem in version 1.24.11, python3.8.10

**Comment by yang-521** [comment]
I updated python to 3.13.2 and pymupdf to the latest version, but this problem still exists. It is present in all pdf files that have inserted text using pymupdf. Before the text was inserted, acrobat could edit normally. I have also tried editing with other versions of acrobat. But the same problem exists........ Very strange. Is this a problem of the computer environment? @JorjMcKie

**Reply by JorjMcKie** [reply]
I am not able to tell what the problem of Adobe Acrobat may have.
The text inserted with PyMuPDF follows the standard PDF specifications, can be displayed by all PDF viewers and likewise can be extracted by relevant software.
So I am sorry to say that I have no advice on how to proceed.

**Comment by yang-521** [comment]
Most PDF editors have no problem opening PDF files with text inserted using pymupdf, except Acrobat. Could it be that the PDF standards used are inconsistent? I tried adding text using PDF-xchange and then opening it with Acrobat, but this issue did not occur.

**Reply by JorjMcKie** [reply]
You cannot add standard text with PDF-xchange (the free version at least): you probably used one of these tools
![image](https://github.com/user-attachments/assets/01b5f9ef-1a5a-4ecc-9fae-2c315f70636f)
That's not what we are talking about here.

**Comment by yang-521** [comment]
I conducted many tests and found that this problem would not occur if the default helv was used instead of using insert_font to insert the font. But when I output files using cjk, acrobat will go wrong. The following are my files and code segments.
```python
import fitz  # PyMuPDF
#
doc = fitz.open('1.pdf')
page = doc.load_page(0) 
font = fitz.Font(fontname="cjk")  
page.insert_font(fontname="cjk", fontbuffer=font.buffer)  
page.insert_text((50, 50),
                "text1",
                fontsize=20,
                fontname="cjk",
                 )
doc.save('3.pdf',garbage=3,deflate=True)
``` 
1 is the original file, 2 is the file using helv, and 3 is the file using cjk. 3 will cause acrobat to report an error, but 2 won't
[1.pdf](https://github.com/user-attachments/files/20256153/1.pdf)
[2.pdf](https://github.com/user-attachments/files/20256154/2.pdf)
[3.pdf](https://github.com/user-attachments/files/20256155/3.pdf)

**Comment by yang-521** [comment]
And when the PDF output using the morph parameter is converted using Ghostscript, such an error will occur
![image](https://github.com/user-attachments/assets/4561dcdd-61a5-4560-8bc6-7f9e760d3701)
[Uploading 4.pdf…]()

**Reply by JorjMcKie** [reply]
What "morph" parameter?
I see no indication for any problem. MuPDF's CLI tool can read it, compress it or build subset fonts, etc. The usual text extraction packages also have no problem at all.
I also looked at the code in the page's `/Content` and manually checked the coding of the inserted text.

So, as long as Adobe cannot tell us details about what it thinks is wrong ("unknown error" is a ridiculous message) there is no way to do anything here.

**Comment by yang-521** [comment]
@JorjMcKie

**Comment by yang-521** [comment]
So, perhaps the inserted text does have problems indeed

**Comment by yang-521** [comment]
The fourth file was not uploaded successfully,This is the file after adding the morph parameter.
[4.pdf](https://github.com/user-attachments/files/20260230/4.pdf)
![image](https://github.com/user-attachments/assets/74fd231a-f751-4dba-a53f-7c30bd457565)
This error occurred when using Ghostscript for processing
![image](https://github.com/user-attachments/assets/3ea41edb-4345-4aa0-b974-e293b73998e0)
It doesn't conform to adobe's PDF specification. This is not a reminder from adobe

**Comment by yang-521** [comment]
However, when this parameter uses the helv font and inserts text without using insert_font, there will be no error. So, could it be a problem with insert_font?

**Reply by JorjMcKie** [reply]
No.

**Comment by Bkhxq** [comment]
i also have the problem. my code is :
```
doc = pymupdf.open(template_path)
        page = doc[0]
        css = """
@font-face {
    font-family: SimSun;
    src: url(simsun.ttc);
    font-weight: normal;
}
* {font-family: SimSun,sans-serif;text-align: justify;}
"""
        archive = pymupdf.Archive(os.path.dirname(template_path))
        rect = pymupdf.Rect([408, 625, 490, 680])
        page.clean_contents()
        page.insert_htmlbox(
            rect,
            f"<p style='font-size:14px;'>{notice_date}</p>",
            css=css,
            archive=archive,
        )
        doc.subset_fonts()
        doc.ez_save(output_path)
        return output_path
```
<img width="2013" height="313" alt="image" src="https://github.com/user-attachments/assets/ef0fa3d6-53d2-4a1f-8f0c-d4afd93c7e56" />

**Comment by Sparrow2333** [comment]
@yang-521 @Bkhxq
This may be related to the version of Adobe Acrobat.

Python version: 3.11
PyMuPDF version: 1.26.6

The same code was executed under the above environment on both macOS and Linux, producing identical results:

* **Adobe Acrobat 2025.001.20623**: The editing feature works normally.
* **Adobe Acrobat Pro 2023.003.20244**: The editing feature works normally.
* **Adobe Acrobat Pro DC 2019.008.20074**: An **“Unknown Error”** occurs when using the editing feature.
